### DISCUSSION TITLE
Using attributes in python

### DISCUSSION THREAD

**Discussion by Mohao3issi** [community/community#21436] [Other Feature Feedback, Questions, & Ideas]
<p>Hello everyone ,<br>
Please what is the difference between when naming an attribute by self.message and when naming it by message as an argument of a method like here:</p>
<pre><code class="lang-auto">def talk(self,message)
    print({} sad : {}".format(self.name,message))
</code></pre>
<p>second method</p>
<pre><code class="lang-auto">def talk(self,message)
    print({} sad : {}".format(self.name,self.message))
</code></pre>
<p>and thank you</p>

**Comment by airtower-luna** [comment]
<p>They are completely different things:</p>
<ul>
<li>
<p><code>self.message</code> reads an attribute of the <code>self</code> object (or tries to, it will fail if the object doesn’t have such an attribute).</p>
</li>
<li>
<p>If you’re passing a <code>message</code> parameter to a function and reference that, whatever has been passed in is used.</p>
</li>
</ul>
<p>Which makes sense to do depends on what you’re actually doing. Using <code>self</code> sounds like you’re learning about objects. :slightly_smiling_face:</p>

**Comment by Mohao3issi** [comment]
<p>Please can you more explain i didn’t understand yet</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>If you have more questions, please ask them specifically. Or check the Python tutorial chapter on classes and objects, it’s pretty extensive: <a href="https://docs.python.org/3/tutorial/classes.html" class="inline-onebox">9. Classes — Python 3.9.7 documentation</a></p>
